amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -476,6 +477,25 @@ void runProcessLoop() {
                     }
                 };
 
+                if (formerStreamsLeasesDeletionStrategy.shouldCleanupLeasesForDeletedStreams()) {
+                    // We do lease sync for old streams, before leaving to the deletion strategy to delete leases for
+                    // strategy detected leases. Also, for deleted streams we expect the shard sync to remove the
+                    // leases.
+                    Iterator<StreamIdentifier> currentSetOfStreamsIter = currentStreamConfigMap.keySet().iterator();
```

Comment:
nit: might be able to shorten
```java
currentStreamConfigMap.keySet()
        .stream()
        .filter(streamIdentifier -> !newStreamConfigMap.containsKey(streamIdentifier))
        .forEach(streamIdentifier -> leaseSync(streamsSynced, streamIdentifier));

...


private void leaseSync(Set<StreamIdentifier> streamsSynced, StreamIdentifier streamIdentifier) {
    log.info("Found old/deleted stream: " + streamIdentifier
    + ". Syncing shards of that stream.");
    ShardSyncTaskManager shardSyncTaskManager = createOrGetShardSyncTaskManager(
    currentStreamConfigMap.get(streamIdentifier));
    shardSyncTaskManager.syncShardAndLeaseInfo();
    currentStreamConfigMap.remove(streamIdentifier);
    streamsSynced.add(streamIdentifier);
}
```

Reply:
I just copied this logic from the existing firehose version. Keeping it this way, to close the PR.